projects/plugins/jetpack/_inc/client/components/data/query-site-products/index.jsx

```diff
@@ -0,0 +1,35 @@
+import { isEmpty } from 'lodash';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+import { fetchSiteProducts, isFetchingSiteProducts, getSiteProducts } from 'state/site-products';
+
+class QuerySiteProducts extends Component {
+	static propTypes = {
+		isFetchingSiteProducts: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		isFetchingSiteProducts: false,
+	};
+
+	componentDidMount() {
+		if ( ! this.props.isFetchingSiteProducts && isEmpty( this.props.siteProducts ) ) {
```

Comment:
The call to `isEmpty` means that we'll basically fetch site products only once per user session. I guess this is fine in most cases, but could potentially lead to some discrepancies between the plugin and server state. Is this a common pattern in the plugin codebase?

In Calypso, we've been trying to stop using `lodash` (I think it has been removed by now). I don't know if we're trying to do the same with the plugin.

Reply:
@monsieur-z I copied essentially this exact pattern from the other data query components in the `components/data` directory. If we are trying to get rid of it in the plugin, I am thinking maybe stick with consistency for now and then go through all of these and edit them all at once at some point to remove the use of lodash?